lib/objects.rb

```diff
@@ -131,10 +132,26 @@ def send_media_object(object, new_object, old_pid=nil)
                     avalon_url: "#{routing_target[:url]}/media_objects/#{pid}",
                     message: 'successfully deposited in avalon'
                   }
-    update_info[:api_hash] = payload if old_pid.present?
     update_status(object[:json][:group_name], update_info)
   end
 
+  # If object was previously posted, save the original avalon pid to the identifier field
+  # @param [Hash] object the object submitted to Switchyard
+  # @param [String] the pid from when object was first posted to avalon
+  # @return [String] the pid to send with the post, or nil
+  def save_old_pid_in_hash!(object, old_pid)
+    if old_pid.present?
+      js = JSON.parse(@posted_content)
+      unless js['metadata']['identifier'].present?
```

Comment:
This guard is good until we change ids again but we can deal with that at that time. 😄